src/program.ts

```diff
@@ -3783,7 +3783,7 @@ export class Function extends TypedElement {
         let scopedLocals = this.flow.scopedLocals;
         if (!scopedLocals) this.flow.scopedLocals = scopedLocals = new Map();
         scopedLocals.set(CommonNames.this_, local);
-        this.localsByIndex[local.index] = local;
+        this.localsByIndex.push(local);
```

Comment:
Pushing to `localsByIndex` feels a bit odd. Using indexed assignment here (and below) would be clearer I think?